### ISSUE TITLE
Fail mapping file to json.

### ISSUE THREAD

**Issue by BOFA1ex** [GoodforGod/DummyMapper#27]
**Stack-trace**
Stack-trace from IDEA Event Log.

```log
There was an error mapping file to JSON. Please report this issue here: https://github.com/GoodforGod/DummyMapper/issues Stacktrace: java.lang.IllegalArgumentException: Unexpected IOException reading JAR File '/Users/bofa1ex/Library/Application%20Support/JetBrains/IntelliJIdea2020.2/plugins/dummymapper/lib/dummymaker-3.0.1.jar' at io.dummymaker.scan.impl.ResourceScanner.loadFromJar(ResourceScanner.java:72) at io.dummymaker.scan.impl.ResourceScanner.lambda$scan$0(ResourceScanner.java:92) at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195) at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1655) at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484) at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474) at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913) at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578) at io.dummymaker.scan.impl.ResourceScanner.scan(ResourceScanner.java:95) at io.dummymaker.scan.impl.ClassScanner.scan(ClassScanner.java:22) at io.dummymaker.factory.impl.GenSupplier.getClassifiedGenerators(GenSupplier.java:149) at io.dummymaker.factory.impl.GenSupplier. (GenSupplier.java:49) at io.dummymaker.factory.impl.GenFactory. (GenFactory.java:67) at io.goodforgod.dummymapper.service.GenFactoryProvider.get(GenFactoryProvider.java:47) at io.goodforgod.dummymapper.mapper.impl.JsonMapper.map(JsonMapper.java:57) at io.goodforgod.dummymapper.mapper.impl.JsonMapper.map(JsonMapper.java:70) at io.goodforgod.dummymapper.MapperAction.actionPerformed(MapperAction.java:104) at com.intellij.openapi.actionSystem.ex.ActionUtil.performActionDumbAware(ActionUtil.java:282) at com.intellij.openapi.actionSystem.impl.ActionMenuItem$ActionTransmitter.lambda$actionPerformed$0(ActionMenuItem.java:296) at com.intellij.openapi.wm.impl.FocusManagerImpl.runOnOwnContext(FocusManagerImpl.java:281) at com.intellij.openapi.wm.impl.IdeFocusManagerImpl.runOnOwnContext(IdeFocusManagerImpl.java:77) at com.intellij.openapi.actionSystem.impl.ActionMenuItem$ActionTransmitter.actionPerformed(ActionMenuItem.java:285) at java.desktop/javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:1967) at com.intellij.openapi.actionSystem.impl.ActionMenuItem.lambda$fireActionPerformed$0(ActionMenuItem.java:112) at com.intellij.openapi.application.TransactionGuardImpl.performUserActivity(TransactionGuardImpl.java:94) at com.intellij.openapi.actionSystem.impl.ActionMenuItem.fireActionPerformed(ActionMenuItem.java:112) at com.intellij.ui.plaf.beg.BegMenuItemUI.doClick(BegMenuItemUI.java:519) at com.intellij.ui.plaf.beg.BegMenuItemUI.access$300(BegMenuItemUI.java:36) at com.intellij.ui.plaf.beg.BegMenuItemUI$MyMouseInputHandler.mouseReleased(BegMenuItemUI.java:541) at java.desktop/java.awt.Component.processMouseEvent(Component.java:6654) at java.desktop/javax.swing.JComponent.processMouseEvent(JComponent.java:3345) at java.desktop/java.awt.Component.processEvent(Component.java:6419) at java.desktop/java.awt.Container.processEvent(Container.java:2263) at java.desktop/java.awt.Component.dispatchEventImpl(Component.java:5029) at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2321) at java.desktop/java.awt.Component.dispatchEvent(Component.java:4861) at java.desktop/java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4918) at java.desktop/java.awt.LightweightDispatcher.processMouseEvent(Container.java:4547) at java.desktop/java.awt.LightweightDispatcher.dispatchEvent(Container.java:4488) at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2307) at java.desktop/java.awt.Window.dispatchEventImpl(Window.java:2773) at java.desktop/java.awt.Component.dispatchEvent(Component.java:4861) at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:778) at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:727) at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721) at java.base/java.security.AccessController.doPrivileged(Native Method) at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85) at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:95) at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:751) at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:749) at java.base/java.security.AccessController.doPrivileged(Native Method) at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85) at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:748) at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:971) at com.intellij.ide.IdeEventQueue.dispatchMouseEvent(IdeEventQueue.java:906) at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:838) at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$8(IdeEventQueue.java:452) at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:744) at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$9(IdeEventQueue.java:451) at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:802) at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:505) at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203) at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124) at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113) at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109) at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101) at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90) Caused by: java.nio.file.NoSuchFileException: /Users/bofa1ex/Library/Application%20Support/JetBrains/IntelliJIdea2020.2/plugins/dummymapper/lib/dummymaker-3.0.1.jar at java.base/sun.nio.fs.UnixException.translateToIOException(UnixException.java:92) at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:111) at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:116) at java.base/sun.nio.fs.UnixFileAttributeViews$Basic.readAttributes(UnixFileAttributeViews.java:55) at java.base/sun.nio.fs.UnixFileSystemProvider.readAttributes(UnixFileSystemProvider.java:149) at java.base/java.nio.file.Files.readAttributes(Files.java:1763) at java.base/java.util.zip.ZipFile$Source.get(ZipFile.java:1225) at java.base/java.util.zip.ZipFile$CleanableResource. (ZipFile.java:727) at java.base/java.util.zip.ZipFile$CleanableResource.get(ZipFile.java:844) at java.base/java.util.zip.ZipFile. (ZipFile.java:247) at java.base/java.util.zip.ZipFile. (ZipFile.java:177) at java.base/java.util.jar.JarFile. (JarFile.java:348) at java.base/java.util.jar.JarFile. (JarFile.java:319) at java.base/java.util.jar.JarFile. (JarFile.java:258) at io.dummymaker.scan.impl.ResourceScanner.loadFromJar(ResourceScanner.java:65) ... 67 more
```

**Describe the bug**
none.

**Java Class examples \ Screenshots**
```
public class User {

    private class Credential {
        private String id;
        private long issued;
    }

    private UUID id;
    private String name;
    private List<String> roles;
    private Credential credential;
}
```

**Expected Behavior**
mapper to json succeed.

**IDEA version**
IDEA 2020.02

**Comment by GoodforGod** [comment]
Hello, can you check is there actually a JAR present?
And can you run IDEA under root, maybe plugin can not access jar on such path due access absent?

Unfortunately error message from ResourceScanner was not properly thrown to the top, so I not sure I can see real reason here

**Comment by BOFA1ex** [comment]
![image](https://user-images.githubusercontent.com/30573693/94508354-66c2b480-0244-11eb-83e4-efd40242a2f5.png)
dummymapper plugin library was present actually.

![image](https://user-images.githubusercontent.com/30573693/94508634-f49e9f80-0244-11eb-9bcc-69bd767c09f6.png)

i try to run ./idea in terminal, here is the error log in console when i choose the option [**Map as JSON**].

```
2020-09-29 11:06:37,157 [  57591]   WARN - util.DeprecatedMethodException - The default implementation of method 'com.intellij.execution.configurations.ConfigurationFactory.getId' is deprecated, you need to override it in 'class org.jetbrains.kotlin.idea.nodejs.protractor.KotlinProtractorConfigurationType$1'. The default implementation delegates to 'getName' which may be localized but return value of this method must not depend on current localization. 
com.intellij.util.DeprecatedMethodException: The default implementation of method 'com.intellij.execution.configurations.ConfigurationFactory.getId' is deprecated, you need to override it in 'class org.jetbrains.kotlin.idea.nodejs.protractor.KotlinProtractorConfigurationType$1'. The default implementation delegates to 'getName' which may be localized but return value of this method must not depend on current localization.
	at com.intellij.util.DeprecatedMethodException.reportDefaultImplementation(DeprecatedMethodException.java:38)
	at com.intellij.execution.configurations.ConfigurationFactory.getId(ConfigurationFactory.java:80)
	at com.intellij.execution.impl.RunManagerImplKt.getFactoryKey(RunManagerImpl.kt:1313)
	at com.intellij.execution.impl.RunManagerImplKt.access$getFactoryKey(RunManagerImpl.kt:1)
	at com.intellij.execution.impl.RunManagerImpl.getConfigurationTemplate(RunManagerImpl.kt:326)
	at com.intellij.execution.impl.RunManagerImpl.createConfiguration(RunManagerImpl.kt:251)
	at com.intellij.execution.actions.RunConfigurationProducer.cloneTemplateConfigurationStatic(RunConfigurationProducer.java:258)
	at com.intellij.execution.actions.RunConfigurationProducer.cloneTemplateConfiguration(RunConfigurationProducer.java:248)
	at com.intellij.execution.actions.RunConfigurationProducer.createConfigurationFromContext(RunConfigurationProducer.java:97)
	at com.intellij.execution.actions.RunConfigurationProducer.findOrCreateConfigurationFromContext(RunConfigurationProducer.java:198)
	at com.intellij.execution.actions.PreferredProducerFind.doGetConfigurationsFromContext(PreferredProducerFind.java:110)
	at com.intellij.execution.actions.PreferredProducerFind.getConfigurationsFromContext(PreferredProducerFind.java:93)
	at com.intellij.execution.actions.PreferredProducerFind.findConfigurationFromContext(PreferredProducerFind.java:135)
	at com.intellij.execution.actions.PreferredProducerFind.createConfiguration(PreferredProducerFind.java:32)
	at com.intellij.execution.actions.ConfigurationContext.createConfiguration(ConfigurationContext.java:160)
	at com.intellij.execution.actions.ConfigurationContext.getConfiguration(ConfigurationContext.java:151)
	at krasa.mavenhelper.action.RunTestFileAction.update(RunTestFileAction.java:171)
	at com.intellij.openapi.actionSystem.ex.ActionUtil.performDumbAwareUpdate(ActionUtil.java:179)
	at com.intellij.openapi.actionSystem.impl.ActionUpdater.doUpdate(ActionUpdater.java:453)
	at com.intellij.openapi.actionSystem.impl.ActionUpdater.lambda$new$1(ActionUpdater.java:94)
	at com.intellij.openapi.actionSystem.impl.ActionUpdater.callAction(ActionUpdater.java:134)
	at com.intellij.openapi.actionSystem.impl.ActionUpdater.lambda$new$2(ActionUpdater.java:95)
	at com.intellij.openapi.actionSystem.impl.ActionUpdater.update(ActionUpdater.java:435)
	at com.intellij.openapi.actionSystem.impl.ActionUpdater.hasChildrenWithState(ActionUpdater.java:396)
	at com.intellij.openapi.actionSystem.impl.ActionUpdater.hasVisibleChildren(ActionUpdater.java:378)
	at com.intellij.openapi.actionSystem.impl.ActionUpdater.expandGroupChild(ActionUpdater.java:311)
	at com.intellij.openapi.actionSystem.impl.ActionUpdater.lambda$doExpandActionGroup$19(ActionUpdater.java:276)
	at com.intellij.util.containers.ContainerUtil.concat(ContainerUtil.java:1611)
	at com.intellij.openapi.actionSystem.impl.ActionUpdater.doExpandActionGroup(ActionUpdater.java:276)
	at com.intellij.openapi.actionSystem.impl.ActionUpdater.expandGroupChild(ActionUpdater.java:327)
	at com.intellij.openapi.actionSystem.impl.ActionUpdater.lambda$doExpandActionGroup$19(ActionUpdater.java:276)
	at com.intellij.util.containers.ContainerUtil.concat(ContainerUtil.java:1611)
	at com.intellij.openapi.actionSystem.impl.ActionUpdater.doExpandActionGroup(ActionUpdater.java:276)
	at com.intellij.openapi.actionSystem.impl.ActionUpdater.expandGroupChild(ActionUpdater.java:327)
	at com.intellij.openapi.actionSystem.impl.ActionUpdater.lambda$doExpandActionGroup$19(ActionUpdater.java:276)
	at com.intellij.util.containers.ContainerUtil.concat(ContainerUtil.java:1611)
	at com.intellij.openapi.actionSystem.impl.ActionUpdater.doExpandActionGroup(ActionUpdater.java:276)
	at com.intellij.openapi.actionSystem.impl.ActionUpdater.expandActionGroup(ActionUpdater.java:186)
	at com.intellij.openapi.actionSystem.impl.ActionUpdater.expandActionGroup(ActionUpdater.java:160)
	at com.intellij.openapi.actionSystem.impl.ActionUpdater.lambda$expandActionGroupWithTimeout$11(ActionUpdater.java:202)
	at com.intellij.openapi.progress.ProgressManager.lambda$runProcess$0(ProgressManager.java:56)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:170)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:629)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:581)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:60)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:157)
	at com.intellij.openapi.progress.ProgressManager.runProcess(ProgressManager.java:56)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.withTimeout(ProgressIndicatorUtils.java:307)
	at com.intellij.openapi.actionSystem.impl.ActionUpdater.expandActionGroupWithTimeout(ActionUpdater.java:202)
	at com.intellij.openapi.actionSystem.impl.ActionUpdater.expandActionGroupWithTimeout(ActionUpdater.java:194)
	at com.intellij.openapi.actionSystem.impl.Utils.fillMenu(Utils.java:98)
	at com.intellij.openapi.actionSystem.impl.ActionPopupMenuImpl$MyMenu.show(ActionPopupMenuImpl.java:118)
	at com.intellij.openapi.editor.impl.ContextMenuPopupHandler.handlePopup(ContextMenuPopupHandler.java:35)
	at com.intellij.openapi.editor.impl.EditorImpl.invokePopupIfNeeded(EditorImpl.java:4783)
	at com.intellij.openapi.editor.impl.EditorImpl.access$9400(EditorImpl.java:124)
	at com.intellij.openapi.editor.impl.EditorImpl$MyMouseAdapter.runMousePressedCommand(EditorImpl.java:3843)
	at com.intellij.openapi.editor.impl.EditorImpl$MyMouseAdapter.mousePressed(EditorImpl.java:3764)
	at java.desktop/java.awt.Component.processMouseEvent(Component.java:6651)
	at java.desktop/javax.swing.JComponent.processMouseEvent(JComponent.java:3345)
	at java.desktop/java.awt.Component.processEvent(Component.java:6419)
	at java.desktop/java.awt.Container.processEvent(Container.java:2263)
	at java.desktop/java.awt.Component.dispatchEventImpl(Component.java:5029)
	at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2321)
	at java.desktop/java.awt.Component.dispatchEvent(Component.java:4861)
	at java.desktop/java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4918)
	at java.desktop/java.awt.LightweightDispatcher.processMouseEvent(Container.java:4544)
	at java.desktop/java.awt.LightweightDispatcher.dispatchEvent(Container.java:4488)
	at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2307)
	at java.desktop/java.awt.Window.dispatchEventImpl(Window.java:2773)
	at java.desktop/java.awt.Component.dispatchEvent(Component.java:4861)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:778)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:727)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:95)
	at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:751)
	at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:749)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:748)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:971)
	at com.intellij.ide.IdeEventQueue.dispatchMouseEvent(IdeEventQueue.java:906)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:838)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$8(IdeEventQueue.java:452)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:744)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$9(IdeEventQueue.java:451)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:802)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:505)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)
SLF4J: Class path contains multiple SLF4J bindings.
SLF4J: Found binding in 
[jar:file:/Users/bofa1ex/Library/Application%20Support/JetBrains/IntelliJIdea2020.2/plugins/dummymapper/lib/logback-classic-1.2.3.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: Found binding in [jar:file:/Applications/IntelliJ%20IDEA.app/Contents/lib/slf4j-log4j12-1.7.25.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: See http://www.slf4j.org/codes.html#multiple_bindings for an explanation.
SLF4J: Actual binding is of type [ch.qos.logback.classic.util.ContextSelectorStaticBinder]
```

**Comment by deanpullenesure** [comment]
I'm wondering if it's purely down to the whitespace in the directory being handled incorrectly, as I've seen the same issue,

What I see is:

`09/12/2020
14:52	Failed mapping to JSON: There was an error mapping file to JSON.
			Please report this issue here: https://github.com/GoodforGod/DummyMapper/issues
			Stacktrace: java.lang.IllegalArgumentException: Unexpected IOException reading JAR File '/Users/myname/Library/Application%20Support/JetBrains/IntelliJIdea2020.2/plugins/dummymapper/lib/dummymaker-3.0.1.jar'
				at io.dummymaker.scan.impl.ResourceScanner.loadFromJar(ResourceScanner.java:72)
				at io.dummymaker.scan.impl.ResourceScanner.lambda$scan$0(ResourceScanner.java:92)
				at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
				at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1655)
				at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
				at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
				at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.j... (show balloon)
`

i.e. the path: '/Users/myname/Library/Application%20Support/JetBrains/IntelliJIdea2020.2/plugins/dummymapper/lib/dummymaker-3.0.1.jar'
	
Is the %20 support in OSX, it certainly won't work if I try it from the command line, but will if I replace it with a space.

**Comment by GoodforGod** [comment]
Actually that might be (didn't see the white space in path first time)
I'll try to reproduce it on windows\linux for such behaviour.

**Comment by GoodforGod** [comment]
I can build and provide you plugin with potential fix, can you manually install it in IDEA and try to reproduce this issue?

**Comment by GoodforGod** [comment]
Uploaded patched plugin to drive, if you can try to reproduce it or this patch fixes issue, that would be great, couldn't reproduce it on windows or linux.

drive.goo gle.com/file/d/1zonTM62yKqlgYt6jRaH53HVRaVKBqA7a/view?usp=sharing

**Comment by BOFA1ex** [comment]
i found that i could not access to this link ` gle.com/file/d/1zonTM62yKqlgYt6jRaH53HVRaVKBqA7a/view?usp=sharing`

Is there another way to get it?

**Comment by GoodforGod** [comment]
This is link to google drive actually, there was space between link pars, so bots won't parse it
`drive.google.com/file/d/1zonTM62yKqlgYt6jRaH53HVRaVKBqA7a/view?usp=sharing`

**Comment by BOFA1ex** [comment]
```
Failed mapping to JSON: There was an error mapping file to JSON.
Please report this issue here: https://github.com/GoodforGod/DummyMapper/issues
Stacktrace: java.lang.IllegalArgumentException: Exception while reading JAR File '/Users/bofa1ex/Library/Application%20Support/JetBrains/IntelliJIdea2020.2/plugins/dummymapper/lib/dummymaker-3.0.2-SNAPSHOT.jar' with error message: /Users/bofa1ex/Library/Application%20Support/JetBrains/IntelliJIdea2020.2/plugins/dummymapper/lib/dummymaker-3.0.2-SNAPSHOT.jar
at io.dummymaker.scan.impl.ResourceScanner.loadFromJar(ResourceScanner.java:92)
at io.dummymaker.scan.impl.ResourceScanner.lambda$scan$0(ResourceScanner.java:40)
at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1655)
at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
at java.base/java.ut... (show balloon)
```
it appear the same situation.

**Comment by GoodforGod** [comment]
Thanks for checking, i'll try to get with a new potential fix in some hours, looks like path need to be properly decoded %20 symbol, though that was issue in my side first, but looks like thats how the path is received by library

I'll post a patched plugin when I'll figure out a solution, thanks again!

**Comment by GoodforGod** [comment]
Updated path normalization via JDK utils, hope this would help in this case
Here is patched plugin
`drive.google.com/file/d/1mZlHGsqP1ebuthKGwzXDjIqOOAq5mPSb/view?usp=sharing`

**Comment by BOFA1ex** [comment]
```
Failed mapping to JSON: There was an error mapping file to JSON.
Please report this issue here: https://github.com/GoodforGod/DummyMapper/issues
Stacktrace: java.lang.IllegalArgumentException: Can not open JAR 'jar:file:/Users/bofa1ex/Library/Application%20Support/JetBrains/IntelliJIdea2020.2/plugins/dummymapper/lib/dummymaker-3.0.2-SNAPSHOT.jar!/io/dummymaker/generator/', failed with: Users/bofa1ex/Library/Application%20Support/JetBrains/IntelliJIdea2020.2/plugins/dummymapper/lib/dummymaker-3.0.2-SNAPSHOT.jar
at io.dummymaker.scan.impl.ResourceScanner.loadFromJar(ResourceScanner.java:101)
at io.dummymaker.scan.impl.ResourceScanner.lambda$scan$0(ResourceScanner.java:42)
at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1655)
at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
at java... (show balloon)
```

i saw the ResourceScanner#loadFromJar code, maybe u can try this solution in below.
it seems work when i try it in local mode.

comment this line `String canonical = jarPath.startsWith("/") ? jarPath.substring(1) : jarPath;`
and then append this code`jarPath = URLDecoder.decode(jarPath, Charsets.UTF_8.name());`
in try/catch statement 

![image](https://user-images.githubusercontent.com/30573693/102436281-2d376380-4053-11eb-94b6-5d63fd76275b.png)

**Comment by GoodforGod** [comment]
Canonical path looks useless on Unix, but useless on windows, removed that part
Solution with Decoder looks like a proper one, hope this will resolve issue, thanks for his!

Can you please try patched plugin one more time and hope this will resolve issue
`drive.google.com/file/d/1uRCwCW7rfJBTVVWy2yHCQYbgJZkC-FNn/view?usp=sharing`

**Comment by BOFA1ex** [comment]
it work! bro
thanks for ur help!

**Comment by GoodforGod** [comment]
Fix will be available in 1.0.1 release soon.
